### PR TITLE
Remember upload metadata even if state machine changed

### PR DESCRIPTION
**Pull request by mbarton**
tl;dr - we can now change the upload pipeline without losing metadata (user and filename) about each upload.

The Cloud Formation support for Step Functions will delete the old state machine rather than modify it. We rely on this `Upload` data to show the user and filename for each upload in the assets page.

This PR adds a dynamo table to cache the `Upload` data. It is written to at the start and end of each run of the state machine. The upload controller will then look in the cache first **for metadata only**. This does not affect looking up the progress of an upload which is still read from the state machine.

To support this, I've added  `startTimestamp` to `Upload`. This was originally read directly from the state machine execution and is backfilled from there if not present.

TODO:
- [x] ~save executions from CODE~
- [x] deploy CFN to CODE
- [x] ~upload executions to new cache table in CODE~
- [x] test in CODE
- [x] save executions from PROD
- [ ] deploy pipeline CFN to PROD
- [ ] deploy app CFN to PROD
- [ ] merge and deploy to PROD
- [ ] upload executions to new cache table in PROD

### CODE REVIEW

File: app/util/UploadDecorator.scala [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+  def addMetadata(atomId: String, video: ClientAsset): ClientAsset = {
+    val id = s"$atomId-${video.id}"
+
+    getUpload(id) match {
+      case Some(upload) =>
+        video.copy(metadata = Some(
+          ClientAssetMetadata(
+            upload.metadata.originalFilename,
+            upload.metadata.startTimestamp.getOrElse(0),
```
Comment (akash1810) [inline app/util/UploadDecorator.scala]: `0`? 😱

Comment (mbarton) [reply app/util/UploadDecorator.scala]: Yeah this is laziness on my part. It happens to not render in the UI (probably due to an `if(...)` check somewhere) but I should actually make that field optional

**Comment by mbarton** [comment]
@akash1810 I'll merge this Monday morning since it's slightly involved
